File: arrInter.py
class Solution:
    def intersect(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        n1 = sorted(nums1)
        n2 = sorted(nums2)
        intersection = []
        len1 = len(nums1)
        len2 = len(nums2)
        a,b = 0, 0
        # print(n1, "with length", len(nums1))
        # print(n2, "with length", len(nums2))
        while a < len1 and b < len2:
            if n1[a] < n2[b]:
                # print("increasing a")
                a +=1
            elif n2[b] < n1[a]:
                # print("increasing b")
                b +=1
            else:
                # print("adding value %s to intersecrtion array"%(n1[a]))
                intersection.append(n1[a])
                a +=1
                b +=1
            # print(intersection)
        return intersection
                
        

File: arrInter.swift
// Given two arrays, write a function to compute their intersection.

class Solution {
    func intersect(_ nums1: [Int], _ nums2: [Int]) -> [Int] {
        // Counting length of each input arrays
        let l1 = nums1.count
        let l2 = nums2.count
        
        // Create an Empty Intersection Array
        var intersection: [Int] = []
        
        // Sorting input arrays
        var a1 = nums1.sorted()
        var a2 = nums2.sorted()
        
        // Create counter for each arrays
        var c1 = 0, c2 = 0
        
        // Iterate over the array
        while c1 < l1 && c2 < l2 {
            if a1[c1] < a2[c2] {
                c1 += 1
            } else if a1[c1] > a2[c2] {
                c2 += 1
            } else {
                intersection.append(a1[c1])
                c1 += 1
                c2 += 1
            }
        }
        
        // When there is a common number in the array then add the number into the array
        return intersection
    }
}